src/multi/bmp180.cpp

```diff
@@ -196,97 +191,77 @@ bmp180::bmp180(hal::i2c& p_i2c, oversampling_rate p_oversampling_setting)
 
 hal::celsius bmp180::temperature()
 {
-  // run helper function which computes the temperature and b5
+  // Run helper function which computes the temperature and b5.
   auto temperature_data = compute_temperature_and_b5(m_i2c,
                                                      m_address,
                                                      m_calibration_data.ac5,
                                                      m_calibration_data.ac6,
                                                      m_calibration_data.mc,
                                                      m_calibration_data.md);
 
-  // return just the temperature data
+  // Return just the temperature data.
   return temperature_data.temperature;
 }
 
 bmp180::pressure_results bmp180::pressure(int sample_amount)
 {
-  // clamp sample amount to determined maximum
-  sample_amount = std::clamp(sample_amount, 1, m_maximum_samples);
-
-  computation_variables variables;
-  // run helper function which computes the temperature and b5
+  // Run helper function which computes the temperature and b5.
   auto temperature_data = compute_temperature_and_b5(m_i2c,
                                                      m_address,
                                                      m_calibration_data.ac5,
                                                      m_calibration_data.ac6,
                                                      m_calibration_data.mc,
                                                      m_calibration_data.md);
 
-  // tell sensor to start conversion for pressure
-  hal::byte control_setting =
-    (static_cast<hal::byte>(m_oversampling_setting) << 6) | 0x34;
+  // Prepare buffer to configure the sensor for pressure data gathering.
+  std::int16_t oversampling_setting = hal::value(m_oversampling_setting);
+  hal::byte control_setting = (oversampling_setting << 6) | 0x34;
```

Comment:
```suggestion
  hal::byte control_setting = (oversampling_setting << 6) | 0x34;
```

Why do we OR with 0x34? If this is something in the datasheet, its important to call it out so its obvious to the reader.